geowebcache/kml/src/main/java/org/geowebcache/service/kml/KMLService.java

```diff
@@ -235,6 +239,14 @@ public void handleRequest(Conveyor conv) throws GeoWebCacheException {
             }
         }
         tile.setTileLayer(layer);
+        
+        if(Objects.nonNull(tile.getLayer())) {
+            if(tile.getTileIndex()[2]<0) {
+                secDispatcher.checkSecurity(tile.getLayer(), null, null);
```

Comment:
As a follow-up to the above, why do we not need the bounding box / srs here?

Seems like it gets calculated by SecurityDispatcher if we are in the else clause

Reply:
The problem is that the z index is a negative number which `boundsFromIndex` doesn't like.